hal/src/hal-sdcard.adb

```diff
@@ -0,0 +1,382 @@
+with Ada.Text_IO; use Ada.Text_IO;
```

Comment:
This is not an abstraction, it's an implementation. The HAL package should only define ways to communicate with the sdcard, the rest should be in the service layer.

BTW, is SDCard the best name for this? It's more SDC/MMC protocol.